update mkimage-yum.sh to support dnf

ekuric opened the pull request:
Updated mkimage-yum.sh to to support dnf - for Fedora users, currently mkimage-yum.sh only supports yum. Fedora 22 and never use dnf

ekuric committed:
Added support for dnf ( Fedora case )

ekuric committed:
Added support for dnf ( Fedora case )

Signed-off-by: Elvir Kuric <ekuric@redhat.com>

ekuric committed:
Added support for dnf ( Fedora case )

if [ -e ...] traslated to if [ -f ... ]

modified mkimage-yum.sh per @tianon proposal/request

Signed-off-by: Elvir Kuric <elvirkuric@gmail.com>

ekuric committed:
if [ -e ...] traslated to if [ -f ... ]

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "master" git@github.com:ekuric/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~2
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

tianon commented on contrib/mkimage-yum.sh:
```diff
@@ -11,13 +11,18 @@ usage() {
-yum_config=/etc/yum.conf
+
+if [ -f /etc/dnf/dnf.conf ] ; then
+	yum_config=/etc/dnf/dnf.conf
```
Couldn't we just add `alias yum=dnf` here and skip the `if` down below?

ekuric commented:
@tianon hm... not sure.But, how do you mean, where in mkimage-yum.sh to do this? I think yum and dnf does not follow same /etc structure so we need if/else, but if there is better approach, please apply it.

thaJeztah commented:
@ekuric could you also make sure that your commit(s) are signed of properly?

ekuric commented:
@thaJeztah yes, I forgot that last time will respect that rule in future.Thank you for reminder.

thaJeztah commented:
ping @tianon could you explain how you wanted to do https://github.com/docker/docker/pull/17362#discussion_r43017450?

(thanks for fixing the sign-off @ekuric )

LK4D4 commented:
ping @tianon

tianon commented on contrib/mkimage-yum.sh:
```diff
@@ -63,9 +68,15 @@ if [ -d /etc/yum/vars ]; then
+else
+	yum -c "$yum_config" --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
+    		--setopt=group_package_types=mandatory -y groupinstall Core
+	yum -c "$yum_config" --installroot="$target" -y clean all
```
What I meant with my previous comment was that the two code blocks in this `if` statement are _identical_ sans the change from `yum` to `dnf`, so I'd much rather see us do this check up higher and either have a variable of `yum=dnf` and use `$yum` here, or just do `alias yum=dnf` above so that `yum` isn't invoked.

ie:
```bash
yum_config=/etc/yum.conf
if [ -f /etc/dnf/dnf.conf ] && command -v dnf &> /dev/null; then
	yum_config=/etc/dnf/dnf.conf
	alias yum=dnf
fi

# ...

yum -c "$yum_config" --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
	--setopt=group_package_types=mandatory -y groupinstall Core
yum -c "$yum_config" --installroot="$target" -y clean all
```

ekuric committed:
Merge branch 'master' of https://github.com/ekuric/docker

Signed-off-by: Elvir Kuric <elvirkuric@gmail.com>

GordonTheTurtle commented:
Please sign your commits following these rules:
https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work
The easiest way to do this is to amend the last commit:
~~~console
$ git clone -b "master" git@github.com:ekuric/docker.git somewhere
$ cd somewhere
$ git rebase -i HEAD~5
editor opens
change each 'pick' to 'edit'
save the file and quit
$ git commit --amend -s --no-edit
$ git rebase --continue # and repeat the amend for each commit
$ git push -f
~~~

Ammending updates the existing PR. You **DO NOT** need to open a new one.

ekuric commented:
@tianon @thaJeztah I changed file per last @tianon  proposal / advice. Thank you

ekuric committed:
modified mkimage-yum.sh per @tianon proposal/request

Signed-off-by: Elvir Kuric <elvirkuric@gmail.com>

thaJeztah commented on contrib/mkimage-yum.sh:
```diff
@@ -64,7 +70,7 @@ if [ -d /etc/yum/vars ]; then
 
 yum -c "$yum_config" --installroot="$target" --releasever=/ --setopt=tsflags=nodocs \
-    --setopt=group_package_types=mandatory -y groupinstall Core
+    		--setopt=group_package_types=mandatory -y groupinstall Core
```
nit: unrelated change?

thaJeztah commented:
@ekuric could you squash your commits so that there's only a single commit in this PR, and make sure the commit is signed-off? If you need help with that; there's a section in our contribution guide explaining how to; http://docs.docker.com/opensource/workflow/work-issue/#pull-and-rebase-frequently

LK4D4 commented:
ping @ekuric

ekuric committed:
Merge branch 'master' of https://github.com/ekuric/docker

ekuric commented:
@LK4D4 @thaJeztah only 4288fc2 is valid, I tried to clean up this git stuff, but there are stil 73195f0, e7b9a65, e07a78c which I do not want there, but not sure which git magic will clean up them. 
commit 4288fc2 has all what @tianon proposed regarding mkimage-yum.sh, and we just need it,  ... sorry and thank you

thaJeztah commented:
@ekuric feel free to ask for assistance in the #docker-dev IRC channel. I'm not there at the moment, but I'm sure someone is there to help out

LK4D4 commented:
@ekuric You can use `git rebase -i` to remove unwanted commit. https://help.github.com/articles/about-git-rebase/
You need to rebase against commit in same branch(last before your changes).

ekuric committed:
Added support for dnf ( Fedora case )

made contrib/mkimage-yum.sh to be per @tianon proposals
reset merge and did git commit cleaning

Signed-off-by: Elvir Kuric <elvirkuric@gmail.com>

updated mkimage-yum.sh to support distributions which use dnf ( eg : Fedora )

Signed-off-by: Elvir Kuric <elvirkuric@gmail.com>

ekuric commented:
@thaJeztah I guess easier to close this one,all work we had here is in #18142.Thank you!

thaJeztah commented:
Thanks @ekuric !
